From a8d2d58ce5cb1ad1c55344fdc2c9303727a71244 Mon Sep 17 00:00:00 2001
From: Gary Zhang <b13634@freescale.com>
Date: Fri, 11 Nov 2011 13:19:13 +0800
Subject: ENGR00161608 SSI: support 24bit samplerate

Add 24bit sample rate playback support for SSI and sgtl5000
codec driver.

Signed-off-by: Gary Zhang <b13634@freescale.com>
---
 sound/soc/imx/imx-ssi.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/sound/soc/imx/imx-ssi.c b/sound/soc/imx/imx-ssi.c
index 5adc94b..c67c20f 100644
--- a/sound/soc/imx/imx-ssi.c
+++ b/sound/soc/imx/imx-ssi.c
@@ -53,6 +53,9 @@
 #include "imx-ssi.h"
 
 #define SSI_SACNT_DEFAULT (SSI_SACNT_AC97EN | SSI_SACNT_FV)
+#define IMX_SSI_FORMATS \
+	(SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE | \
+	SNDRV_PCM_FMTBIT_S24_LE)
 
 /*
  * SSI Network Mode or TDM slots configuration.
@@ -465,13 +468,13 @@ static struct snd_soc_dai_driver imx_ssi_dai = {
 		.channels_min = 1,
 		.channels_max = 2,
 		.rates = SNDRV_PCM_RATE_8000_96000,
-		.formats = SNDRV_PCM_FMTBIT_S16_LE,
+		.formats = IMX_SSI_FORMATS,
 	},
 	.capture = {
 		.channels_min = 1,
 		.channels_max = 2,
 		.rates = SNDRV_PCM_RATE_8000_96000,
-		.formats = SNDRV_PCM_FMTBIT_S16_LE,
+		.formats = IMX_SSI_FORMATS,
 	},
 	.ops = &imx_ssi_pcm_dai_ops,
 };
-- 
1.7.9.5

